test/python/quantum_info/states/test_statevector.py

```diff
@@ -335,6 +344,20 @@ def test_tensor(self):
             self.assertEqual(state.dims(), (3, 2))
             assert_allclose(state.data, target)
 
+    def test_inner(self):
+        """Test inner method."""
+        for _ in range(10):
+            seed = np.random.randint(0, np.iinfo(np.int32).max)
+            rng = np.random.default_rng(seed)
+            dim = 2 ** rng.integers(1, 4)
+            vec0 = Statevector(self.rand_vec(dim))
+            vec1 = Statevector(self.rand_vec(dim))
+            target = 0
+            for i in range(vec0.dim):
+                target += np.conj(vec0.data[i]) * vec1.data[i]
+            result = vec0.inner(vec1)
+            assert_almost_equal(result, target)
```

Comment:
`unittest` has `self.assertAlmostEqual` for comparing floats - using `numpy` kind of makes it seem like you're expecting an array output here.

Reply:
Nice catch
```suggestion
            self.assertAlmostEqual(result, target)
```